libbpf-tools/tcprtt.bpf.c

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2021 Wenbo Zhang
+#include <vmlinux.h>
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include <bpf/bpf_endian.h>
+#include "tcprtt.h"
+#include "bits.bpf.h"
+#include "maps.bpf.h"
+
+const volatile bool targ_laddr_hist = false;
+const volatile bool targ_raddr_hist = false;
+const volatile bool targ_show_ext = false;
+const volatile __u16 targ_sport = 0;
+const volatile __u16 targ_dport = 0;
+const volatile __u32 targ_saddr = 0;
+const volatile __u32 targ_daddr = 0;
+const volatile bool targ_ms = false;
+
+#define MAX_ENTRIES	10240
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__uint(max_entries, MAX_ENTRIES);
+	__type(key, u64);
+	__type(value, struct hist);
+} hists SEC(".maps");
+
+static struct hist zero;
+
+SEC("fentry/tcp_rcv_established")
+int BPF_PROG(tcp_rcv, struct sock *sk)
+{
+	const struct inet_sock *inet = (struct inet_sock *)(sk);
```

Comment:
Hi @anakryiko I found this not wor, so do we need a helper  `bpf_skc_to_inet_sock` like [bpf_skc_to_tcp_sock](https://github.com/torvalds/linux/blob/dbe69e43372212527abf48609aba7fc39a6daa27/net/core/filter.c#L10536)